### ISSUE TITLE
"Exporting repository manifest" hangs forever

### ISSUE THREAD

**Issue by c-vinet** [cvmfs/cvmfs#3134]
Hello,
We have recently found a strange issue on our cvmfs publishers. It happens when we try to publish a python virtual environment (even an empty one).
The publishing hangs on "Exporting repository manifest"  forever, until we have to cancel it with Ctrl+C.
It looks like this was not happening on cvmfs 2.9.4. 

Do you have any idea what might be causing that hang?

Our setup:
Ubuntu 20.04 on x86_64 servers
cvmfs  2.10.0~1+ubuntu20.04

Command line:

```
cvmfs-tomo@scisoft10:/cvmfs/tomo.esrf.fr/tomotools$ cd $HOME; cvmfs_server publish tomo.esrf.fr
Using auto tag 'generic-2022-12-08T13:10:06Z'
Processing changes...
.....
Waiting for upload of files before committing...
Committing file catalogs...
Wait for all uploads to finish
Exporting repository manifest


^CSynchronization failed

Executed Command:
cvmfs_swissknife sync -u /cvmfs/tomo.esrf.fr -s /var/spool/cvmfs/tomo.esrf.fr/scratch/current -c /var/spool/cvmfs/tomo.esrf.fr/rdonly -t /var/spool/cvmfs/tomo.esrf.fr/tmp -b fe88f26404b7eddce3f8750e81b544e83f27fbdb -r gw,/srv/cvmfs/tomo.esrf.fr/data/txn,http://cernvmfs-stratum0:4929/api/v1 -w http://cernvmfs-stratum0/cvmfs/tomo.esrf.fr -o /var/spool/cvmfs/tomo.esrf.fr/tmp/manifest -e sha1 -Z default -C /etc/cvmfs/repositories.d/tomo.esrf.fr/trusted_certs -N tomo.esrf.fr -K /etc/cvmfs/keys/tomo.esrf.fr.pub -L -D generic-2022-12-08T13:10:06Z -H /etc/cvmfs/keys/tomo.esrf.fr.gw -P /var/spool/cvmfs/tomo.esrf.fr/session_token -f overlayfs -p -l 4194304 -a 8388608 -h 16777216 -i
```

Logs on the publisher:
```
Dec 08 14:09:23 scisoft10 cvmfs_publish[705404]: failed to read-ahead /var/spool/cvmfs/tomo.esrf.fr/tmp/reflog.9Na3nx (22)
Dec 08 14:09:23 scisoft10 cvmfs_publish[705404]: failed to read-ahead /var/spool/cvmfs/tomo.esrf.fr/tmp/tags.vXzU3w (22)
Dec 08 14:09:23 scisoft10 cvmfs_publish[705404]: failed to read-ahead /var/spool/cvmfs/tomo.esrf.fr/tmp/catalog.AACS4v (22)
Dec 08 14:09:23 scisoft10 cvmfs_publish[705404]: (tomo.esrf.fr) opened transaction
Dec 08 14:09:23 scisoft10 cvmfs_publish[705404]: Trying to remount /cvmfs/tomo.esrf.fr read/write... success
Dec 08 14:10:06 scisoft10 cvmfs_swissknife[705968]: failed to read-ahead /var/spool/cvmfs/tomo.esrf.fr/tmp/history.oSvK2b (22)
Dec 08 14:10:06 scisoft10 cvmfs_server[706000]: (tomo.esrf.fr) started publishing
Dec 08 14:10:06 scisoft10 cvmfs_swissknife[706002]: Parameter CVMFS_STATISTICS_DB was not set in the repository configuration file. Using default value: /var/spool/cvmfs/tomo.esrf.fr/stats.db
Dec 08 14:10:06 scisoft10 cvmfs_swissknife[706002]: Parameter CVMFS_STATS_DB_DAYS_TO_KEEP was not set in the repository configuration file. Using default value: 365
Dec 08 14:10:06 scisoft10 cvmfs_swissknife[706002]: failed to read-ahead /var/spool/cvmfs/tomo.esrf.fr/stats.db (22)
Dec 08 14:10:06 scisoft10 cvmfs_swissknife[706002]: failed to read-ahead /var/spool/cvmfs/tomo.esrf.fr/tmp/catalog.2SbBdK (22)
Dec 08 14:11:27 scisoft10 cvmfs_swissknife[706006]: watchdog: unexpected termination (2)
                                                    Timestamp: Thu Dec  8 14:11:27 2022
                                                    
                                                     Crash logged also on file: /tmp/cvmfs-swissknife-sync-stacktrace.2022.12.08-13.10.06.706002
Dec 08 14:11:27 scisoft10 cvmfs_server[706175]: (tomo.esrf.fr) failed to publish
Dec 08 14:11:27 scisoft10 cvmfs_server[706178]: (tomo.esrf.fr) failed to publish
```

Logs on the gateway:
```
Dec 08 14:09:23 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"info","component":"http","req_id":"9aba4407-a429-4bc4-ac06-4bef057abe87","req_dt":0.067851,"method":"POST","url":"/api/v1/l>
Dec 08 14:09:23 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"repository_entity","req_id":"9aba4407-a429-4bc4-ac06-4bef057abe87","req_dt":2.242322,"operation":"find_>
Dec 08 14:09:23 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"info","component":"actions","req_id":"9aba4407-a429-4bc4-ac06-4bef057abe87","req_dt":2.360004,"action":"get_repo","outcome">
Dec 08 14:09:23 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"lease_entity","req_id":"9aba4407-a429-4bc4-ac06-4bef057abe87","req_dt":2.785181,"operation":"find_all_b>
Dec 08 14:09:23 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"lease_entity","req_id":"9aba4407-a429-4bc4-ac06-4bef057abe87","req_dt":2.978587,"operation":"delete_all>
Dec 08 14:09:23 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"lease_entity","req_id":"9aba4407-a429-4bc4-ac06-4bef057abe87","req_dt":3.605216,"operation":"create","t>
Dec 08 14:09:23 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"info","component":"actions","req_id":"9aba4407-a429-4bc4-ac06-4bef057abe87","req_dt":18.784932,"action":"new_lease","outcom>
Dec 08 14:09:23 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"info","component":"http","req_id":"9aba4407-a429-4bc4-ac06-4bef057abe87","req_dt":18.952817,"time":"2022-12-08T14:09:23+01:>
Dec 08 14:10:06 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"info","component":"http","req_id":"8aba8e9a-c8bc-4a3f-9faa-995c27dfbb17","req_dt":0.019959,"method":"POST","url":"/api/v1/p>
Dec 08 14:10:06 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"lease_entity","req_id":"8aba8e9a-c8bc-4a3f-9faa-995c27dfbb17","req_dt":1.492066,"operation":"find_by_to>
Dec 08 14:10:06 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"receiver","req_id":"8aba8e9a-c8bc-4a3f-9faa-995c27dfbb17","req_dt":2.685682,"command":"start","time":"2>
Dec 08 14:10:06 cernvmfs-stratum0 cvmfs_receiver_debug[563988]: CVMFS receiver started
Dec 08 14:10:06 cernvmfs-stratum0 cvmfs_receiver_debug[563988]: PayloadProcessor - lease_path: tomo.esrf.fr/tomotools/test_venv, header digest: 28a3c0697facc582524e4f05cf9c39239bb3118f, head>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"receiver","req_id":"8aba8e9a-c8bc-4a3f-9faa-995c27dfbb17","req_dt":1351.811982,"command":"submit payloa>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"worker_pool","req_id":"8aba8e9a-c8bc-4a3f-9faa-995c27dfbb17","req_dt":1351.969674,"worker_id":4,"task_d>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"info","component":"actions","req_id":"8aba8e9a-c8bc-4a3f-9faa-995c27dfbb17","req_dt":1352.033945,"action":"submit_payload",>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"info","component":"http","req_id":"8aba8e9a-c8bc-4a3f-9faa-995c27dfbb17","req_dt":1352.125486,"time":"2022-12-08T14:10:08+0>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"receiver","req_id":"8aba8e9a-c8bc-4a3f-9faa-995c27dfbb17","req_dt":1352.065566,"pipe":"stderr","time":">
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"receiver","req_id":"8aba8e9a-c8bc-4a3f-9faa-995c27dfbb17","req_dt":1352.186091,"pipe":"stdout","time":">
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"receiver","req_id":"8aba8e9a-c8bc-4a3f-9faa-995c27dfbb17","req_dt":1353.505492,"command":"quit","time":>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_receiver_debug[563988]: CVMFS receiver finished
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"info","component":"http","req_id":"fe773ca9-51d8-4f26-b27b-6cee375804d4","req_dt":0.005276,"method":"POST","url":"/api/v1/p>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"lease_entity","req_id":"fe773ca9-51d8-4f26-b27b-6cee375804d4","req_dt":1.392644,"operation":"find_by_to>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"receiver","req_id":"fe773ca9-51d8-4f26-b27b-6cee375804d4","req_dt":2.740204,"command":"start","time":"2>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_receiver_debug[563996]: CVMFS receiver started
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_receiver_debug[563996]: PayloadProcessor - lease_path: tomo.esrf.fr/tomotools/test_venv, header digest: 465089b1f86caed13da4da08b2957ae1928603be, head>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"receiver","req_id":"fe773ca9-51d8-4f26-b27b-6cee375804d4","req_dt":41.516265,"command":"submit payload">
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"worker_pool","req_id":"fe773ca9-51d8-4f26-b27b-6cee375804d4","req_dt":41.568096,"worker_id":5,"task_dt">
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"receiver","req_id":"fe773ca9-51d8-4f26-b27b-6cee375804d4","req_dt":41.62251,"pipe":"stderr","time":"202>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"receiver","req_id":"fe773ca9-51d8-4f26-b27b-6cee375804d4","req_dt":41.644184,"pipe":"stdout","time":"20>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"info","component":"actions","req_id":"fe773ca9-51d8-4f26-b27b-6cee375804d4","req_dt":41.662368,"action":"submit_payload","o>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"info","component":"http","req_id":"fe773ca9-51d8-4f26-b27b-6cee375804d4","req_dt":41.706208,"time":"2022-12-08T14:10:08+01:>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_receiver_debug[563996]: CVMFS receiver finished
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"receiver","req_id":"fe773ca9-51d8-4f26-b27b-6cee375804d4","req_dt":42.842637,"command":"quit","time":"2>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"info","component":"http","req_id":"45bc18b7-f8fd-4ba0-8bf0-91c5c819b5df","req_dt":0.002371,"method":"POST","url":"/api/v1/l>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"lease_entity","req_id":"45bc18b7-f8fd-4ba0-8bf0-91c5c819b5df","req_dt":0.246121,"operation":"find_by_to>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_gateway[557764]: {"level":"debug","component":"receiver","req_id":"45bc18b7-f8fd-4ba0-8bf0-91c5c819b5df","req_dt":0.827522,"command":"start","time":"2>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: CVMFS receiver started
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: CommitProcessor - lease_path: tomo.esrf.fr/tomotools/test_venv, old hash: fe88f26404b7eddce3f8750e81b544e83f27fbdbC, new hash:>
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: CommitProcessor - lease_path: tomo.esrf.fr/tomotools/test_venv, target root hash: fe88f26404b7eddce3f8750e81b544e83f27fbdb
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: CommitProcessor - lease_path: tomo.esrf.fr/tomotools/test_venv, merging catalogs
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: failed to read-ahead /srv/cvmfs/tomo.esrf.fr/data/txn/receiver/commit_processor.EElNsv/catalog.LEQMGy (22)
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: failed to read-ahead /srv/cvmfs/tomo.esrf.fr/data/txn/receiver/commit_processor.K11t9w/catalog.Gx5Vdx (22)
Dec 08 14:10:08 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: failed to read-ahead /srv/cvmfs/tomo.esrf.fr/data/txn/receiver/commit_processor.2db3aw/catalog.lFq3hw (22)
Dec 08 14:10:09 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: failed to read-ahead /srv/cvmfs/tomo.esrf.fr/data/txn/receiver/commit_processor.zKcq3u/history.qgEtGv (22)
Dec 08 14:10:09 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: failed to read-ahead /srv/cvmfs/tomo.esrf.fr/data/txn/receiver/commit_processor.zKcq3u/catalog.o9E19x (22)
Dec 08 14:10:09 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: CommitProcessor - lease_path: tomo.esrf.fr/tomotools/test_venv, signing manifest
Dec 08 14:10:09 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: failed to read-ahead /srv/cvmfs/tomo.esrf.fr/data/txn/receiver/commit_processor.zKcq3u/.cvmfsreflog.KJGDwz (22)
Dec 08 14:10:09 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: CommitProcessor - lease_path: tomo.esrf.fr/tomotools/test_venv, success.
Dec 08 14:10:09 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: CommitProcessor - lease_path: tomo.esrf.fr/tomotools/test_venv, new root hash: ef76dcbc37410f4d85ced34241724b5776b60c77
Dec 08 14:10:09 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: Parameter CVMFS_STATISTICS_DB was not set in the repository configuration file. Using default value: /var/spool/cvmfs/tomo.esr>
Dec 08 14:10:09 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: Parameter CVMFS_STATS_DB_DAYS_TO_KEEP was not set in the repository configuration file. Using default value: 365
Dec 08 14:10:09 cernvmfs-stratum0 cvmfs_receiver_debug[564004]: failed to read-ahead /var/spool/cvmfs/tomo.esrf.fr/stats.db (22)
```

**Comment by jblomer** [comment]
Interesting! Please try to publish in debug mode with `CVMFS_SERVER_DEBUG=3 cvmfs_server publish tomo.esrf.fr`. That will give us more debug output.

If you can, it would also be useful to attach a `gdb` to the hanging swissknife process and to produce a backtrace with the `t a a bt` command.  There are probably two `cvmfs_swissknife` processes at the time of the hang, and the backtrace of both would be useful.

**Comment by c-vinet** [comment]
Hi,
Here are both GDB backtraces and the publishing log.
Thanks for your help !

[Edit jblomer: got the files, removing them from the issue]

**Comment by jblomer** [comment]
Puzzling. I can see that the publishing works until the final commit request is sent to the gateway. It then hangs waiting for a reply.

I don't see any relevant changes between 2.10.0 and 2.9.4.  Perhaps it is this: when the publish command is executed, is the virtual environment still active? If so, please try to exit it before publishing.

NB: unfortunately, I noticed that the GDB backtrace contains the gateway key secret.  I removed the attachment link from the comment. But it's probably still best to change the secret. Sorry about that!

**Comment by c-vinet** [comment]
No problem for the gateway key, we will change it.
We were not activating the python env at all, but maybe the python venv was a false lead. Since yesterday the issue seems to happen with other softwares...
I noticed a strange thing:
If i attach a strace to the receiver process , i can see that it is doing a chmod 644 on all data chunks forever. Is this a normal behavior ?
```
openat(AT_FDCWD, "/srv/cvmfs/sb.esrf.fr/data/txn/chunk.NCcT8k", O_WRONLY) = 10
futex(0x557110f3b8a8, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x557110f3b858, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x557110f3b8ac, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x557110f3b858, FUTEX_WAKE_PRIVATE, 1) = 1
getpid()                                = 439688
openat(AT_FDCWD, "/srv/cvmfs/sb.esrf.fr/data/txn/chunk.nxP6km", O_RDWR|O_CREAT|O_EXCL, 0600) = 10
fchmod(10, 0644)                        = 0
fcntl(10, F_GETFL)                      = 0x8002 (flags O_RDWR|O_LARGEFILE)
close(10)                               = 0
openat(AT_FDCWD, "/srv/cvmfs/sb.esrf.fr/data/txn/chunk.nxP6km", O_WRONLY) = 10
futex(0x557110f3b8a8, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x557110f3b858, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x557110f3b8ac, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x557110f3b858, FUTEX_WAKE_PRIVATE, 1) = 1
getpid()                                = 439688
openat(AT_FDCWD, "/srv/cvmfs/sb.esrf.fr/data/txn/chunk.yYJxgl", O_RDWR|O_CREAT|O_EXCL, 0600) = 10
fchmod(10, 0644)                        = 0
fcntl(10, F_GETFL)                      = 0x8002 (flags O_RDWR|O_LARGEFILE)
close(10)                               = 0
openat(AT_FDCWD, "/srv/cvmfs/sb.esrf.fr/data/txn/chunk.yYJxgl", O_WRONLY) = 10
futex(0x557110f3b8a8, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x557110f3b858, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x557110f3b8ac, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x557110f3b858, FUTEX_WAKE_PRIVATE, 1) = 1
getpid()                                = 439688
openat(AT_FDCWD, "/srv/cvmfs/sb.esrf.fr/data/txn/chunk.6qXazn", O_RDWR|O_CREAT|O_EXCL, 0600) = 10
fchmod(10, 0644)                        = 0
fcntl(10, F_GETFL)                      = 0x8002 (flags O_RDWR|O_LARGEFILE)
close(10)                               = 0
openat(AT_FDCWD, "/srv/cvmfs/sb.esrf.fr/data/txn/chunk.6qXazn", O_WRONLY) = 10
futex(0x557110f3b8a8, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x557110f3b858, FUTEX_WAKE_PRIVATE, 1) = 1
getpid()                                = 439688
openat(AT_FDCWD, "/srv/cvmfs/sb.esrf.fr/data/txn/chunk.0IVvFm", O_RDWR|O_CREAT|O_EXCL, 0600) = 10
fchmod(10, 0644)                        = 0
fcntl(10, F_GETFL)                      = 0x8002 (flags O_RDWR|O_LARGEFILE)
close(10)                               = 0
openat(AT_FDCWD, "/srv/cvmfs/sb.esrf.fr/data/txn/chunk.0IVvFm", O_WRONLY) = 10
futex(0x557110f3b8ac, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x557110f3b858, FUTEX_WAKE_PRIVATE, 1) = 1
getpid()                                = 439688
openat(AT_FDCWD, "/srv/cvmfs/sb.esrf.fr/data/txn/chunk.a0WeMk", O_RDWR|O_CREAT|O_EXCL, 0600) = 10
fchmod(10, 0644)                        = 0
fcntl(10, F_GETFL)                      = 0x8002 (flags O_RDWR|O_LARGEFILE)
close(10)                               = 0
openat(AT_FDCWD, "/srv/cvmfs/sb.esrf.fr/data/txn/chunk.a0WeMk", O_WRONLY) = 10
futex(0x557110f3b8a8, FUTEX_WAKE_PRIVATE, 1) = 1
getpid()                                = 439688
openat(AT_FDCWD, "/srv/cvmfs/sb.esrf.fr/data/txn/chunk.v10tJk", O_RDWR|O_CREAT|O_EXCL, 0600) = 10
fchmod(10, 0644^Cstrace: Process 439688 detached
```

**Comment by jblomer** [comment]
That is expected behavior when the receiver creates temporary files.  It should not be an endless loop though. At some point, the receiver should have processed all the incoming data.  On the gateway, do you see the receiver spinning forever, i.e. being the root cause of the hang?

If you have some instructions of a minimal publish for me that causes the hang, we can try to reproduce it here.

**Comment by jblomer** [comment]
What is indeed strange is that nothing seems to be written into the temporary files. Hmm, another GDB backtrace, this time of the receiver, may bring more clues.

**Comment by c-vinet** [comment]
Hi,
I think we have found the root of the problem.

In short:
- Our cvmfs repository is hosted on an NFS share.
- When we upgraded from 2.9.4 to 2.10, we rebooted the stratum0 and the share was remounted.
- We don't know why, but when the share was remounted it was missing the `nordirplus` mount option.
- Without this option, the crawling of directories with a large number of files is extremely slow and looks like a hang.
- Mounting the repository with `nordirplus` solves completely the issue.

I found the problem by noticing that the cpus on the stratum0 were doing 100% IO during the publishing.
Maybe this is something we can add to the documentation, since some users may end up with the same issue ?

Thanks a lot for your help in debugging this !
